delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+
+    def get_pool_size(self):
+        size_info = self.ssh_do_exec([constant.POOL_SIZE_COMMAND])
+        size_array = size_info.split("\r\n")
+        size_map = {}
+        pool_name = None
+        count = 0
+        for size in size_array:
+            if 'Span ' in size:
+                pool_name = size.split()[-1].replace(':', '')
+                size_map[pool_name] = 0
+                count = 0
+            if '[Free space]' in size:
+                free_array = size.split()
+                if len(free_array) > 2:
+                    count += 1
+                    free_size = free_array[0].replace('GiB', 'GB')
+                    size_map[pool_name] += self.get_size(free_size)
+        return size_map
+
+    def get_pool(self, storage_id):
+        try:
+            pool_info = self.ssh_do_exec([constant.POOL_INFO_COMMAND])
+            pool_list = []
+            pool_array = self.get_table_data(pool_info)
+            size_map = self.get_pool_size()
+            for pool in pool_array:
+                value_array = pool.split()
+                if len(value_array) == 6:
+                    total_capacity = \
+                        self.get_size(value_array[3] + "GB")
+                    free_capacity = \
+                        size_map.get(value_array[0], total_capacity)
+                    status = constants.StoragePoolStatus.NORMAL \
+                        if value_array[1] == 'Yes' \
+                        else constants.StoragePoolStatus.ABNORMAL
+                    pool_model = {
+                        'name': value_array[0],
+                        'storage_id': storage_id,
+                        'native_storage_pool_id': value_array[0],
+                        'description': None,
+                        'status': status,
+                        'storage_type': constants.StorageType.FILE,
+                        'total_capacity': total_capacity,
+                        'used_capacity': total_capacity - free_capacity,
+                        'free_capacity': free_capacity,
+                    }
+                    pool_list.append(pool_model)
+            return pool_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get pool from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get pool from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_controllers(self, storage_id):
+        try:
+            controller_list = []
+            node_info = self.ssh_do_exec([constant.CONTROLLER_INFO_COMMAND])
+            nodes_array = self.get_table_data(node_info)
+            for nodes in nodes_array:
+                node = nodes.split()
+                if len(node) > 2:
```

Comment:
Do not use this magic number
Please investigate all the magic numbers and give it a meaningful name if needed.
Not all the magic number is not allowed, for example:
```
if len(xxx) > 0
```
this is allowed, because we can know what we want to judge whether `xxx` is empty, but for this:
```
if len(xxx) > 3
```
we can know nothing about what this for

Reply:
Numbers are now consolidated in constant files and given names to make them easier to read